openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIHeaderMapper.java

```diff
@@ -0,0 +1,165 @@
+import io.ballerina.compiler.syntax.tree.RequiredParameterNode;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.openapi.converter.Constants;
+import io.ballerina.openapi.converter.utils.ConverterCommonUtils;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.StringSchema;
+import io.swagger.v3.oas.models.parameters.HeaderParameter;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class for the mapping ballerina headers with OAS header parameter sections.
+ *
+ * @since 2.0.0
+ */
+public class OpenAPIHeaderMapper {
+    /**
+     * Handle header parameters in ballerina data type.
+     *
+     * @param headerParam    -  {@link RequiredParameterNode} type header parameter node
+     */
+    public List<Parameter> setHeaderParameter(RequiredParameterNode headerParam) {
+        List<Parameter> parameters = new ArrayList<>();
+        //Handle with string current header a support with only string and string[]
+        String headerName = headerParam.paramName().get().text().replaceAll("\\\\", "");
+        HeaderParameter headerParameter = new HeaderParameter();
+        Node node = headerParam.typeName();
+        StringSchema stringSchema = new StringSchema();
+        NodeList<AnnotationNode> annotations = getAnnotationNodesFromServiceNode(headerParam);
+        String isOptional = Constants.TRUE;
+        if (!annotations.isEmpty()) {
+            isOptional = ConverterCommonUtils.extractServiceAnnotationDetails(annotations,
+                    "http:ServiceConfig", "treatNilableAsOptional");
+        }
+        enableHeaderRequiredOption(headerParameter, node, stringSchema, isOptional);
+        completeHeaderParameter(parameters, headerName, headerParameter, stringSchema, headerParam.annotations(),
+                headerParam.typeName());
+        return parameters;
+    }
+
+    /**
+     * Handle header parameters in ballerina data type.
+     *
+     * @param headerParam    -  {@link DefaultableParameterNode} type header parameter node
+     */
+    public List<Parameter> setHeaderParameter(DefaultableParameterNode headerParam) {
+        List<Parameter> parameters = new ArrayList<>();
+        //Handle with string current header a support with only string and string[]
+        String headerName = headerParam.paramName().get().text().replaceAll("\\\\", "");
+        HeaderParameter headerParameter = new HeaderParameter();
+        StringSchema stringSchema = new StringSchema();
+        stringSchema.setDefault(headerParam.expression().toString().replaceAll("\"", ""));
+        if (headerParam.typeName().kind() == SyntaxKind.OPTIONAL_TYPE_DESC) {
+            stringSchema.setNullable(true);
+        }
```

Comment:
By default, it takes `false`. If make it  `stringSchema.setNullable(false)` it will add a `nullable:false` field to schema but usually we add `nullable` field to schema if it is `true`.  So normally we keep the default value as it is for avoiding adding extra fields to schema. shall we keep this way?

Reply:
Yeah +1